### PR TITLE
[lldb] Implement TypeSystemSwiftTypeRef::CreateTupleType

### PR DESCRIPTION
**Pull request by kastiglione**
Implement `TypeSystemSwiftTypeRef::CreateTupleType`.

rdar://68171404

### CODE REVIEW

File: lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp [hunk 1]
```diff
@@ -1880,10 +1880,46 @@ TypeSystemSwift::TypeAllocationStrategy
...
+      type->addChild(element_type, dem);
+    }
+    return RemangleAsType(dem, tuple);
+  };
+  {
+    auto result = impl();
+    if (!m_swift_ast_context)
+      return result;
+    bool equivalent =
+        Equivalent(result, m_swift_ast_context->CreateTupleType(elements));
```
Comment (adrian-prantl) [inline lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:1923]: Looks good — the only thing that's missing is wrapping the comparison code in #ifndef NDEBUG

**Comment by adrian-prantl** [comment]
And I guess we could unit-test this in the typesystem unit tests?

**Comment by kastiglione** [comment]
sounds good, I'm running existing tests now, I'll add something to the unit tests too.

**Comment by kastiglione** [comment]
For the test, should it construct some types, pass them to `CreateTupleType`, and then test against an expected mangled symbol?

**Comment by adrian-prantl** [comment]
> For the test, should it construct some types, pass them to `CreateTupleType`, and then test against an expected mangled symbol?

Yes. The unit tests are run completely without a SwiftASTContext. Either check for a mangled name directly or expect a specific demangle tree. But I think the name is fine.

**Comment by kastiglione** [comment]
@swift-ci test

**Review by adrian-prantl — APPROVED** [review]

**Pull request merged**